src/operations/search/query/r4.js

```diff
@@ -118,198 +121,32 @@ module.exports.buildR4SearchQuery = (resource_name, args) => {
-            columns.add('location.reference');
-        } else if (['Schedule'].includes(resource_name)) {
-            and_segments.push(referenceQueryBuilder(location_reference, 'actor.reference', location_exists_flag));
-            columns.add('actor.reference');
-        } else {
-            logger.error(`No mapping for searching by location for ${resource_name}: `);
-        }
-    }
-    if (healthcareService || args['healthcareService:missing']) {
-        const healthcareService_reference = 'HealthcareService/' + healthcareService;
-        /**
-         * @type {?boolean}
-         */
-        let healthcareService_exists_flag = null;
-        if (args['healthcareService:missing']) {
-            healthcareService_exists_flag = !isTrue(args['healthcareService:missing']);
-        }
-
-        // each Resource type has a different place to put the patient info
-        if (['HealthcareService'].includes(resource_name)) {
-            query.id = healthcareService;
-            columns.add('id');
-        } else if (['PractitionerRole'].includes(resource_name)) {
-            and_segments.push(referenceQueryBuilder(healthcareService_reference, 'healthcareService.reference', healthcareService_exists_flag));
-            columns.add('healthcareService.reference');
-        } else if (['Schedule'].includes(resource_name)) {
-            and_segments.push(referenceQueryBuilder(healthcareService_reference, 'actor.reference', healthcareService_exists_flag));
-            columns.add('actor.reference');
-        } else {
-            logger.error(`No mapping for searching by healthcareService for ${resource_name}: `);
-        }
-    }
-    if (schedule || args['schedule:missing']) {
-        const schedule_reference = 'Schedule/' + schedule;
-        /**
-         * @type {?boolean}
-         */
-        let schedule_exists_flag = null;
-        if (args['schedule:missing']) {
-            schedule_exists_flag = !isTrue(args['schedule:missing']);
-        }
 
-        // each Resource type has a different place to put the patient info
-        if (['Schedule'].includes(resource_name)) {
-            query.id = schedule;
-            columns.add('id');
-        } else if (['Slot'].includes(resource_name)) {
-            and_segments.push(referenceQueryBuilder(schedule_reference, 'schedule.reference', schedule_exists_flag));
-            columns.add('schedule.reference');
-        } else {
-            logger.error(`No mapping for searching by schedule for ${resource_name}: `);
+    for (const [field, filterObj] of Object.entries(customQueries)) {
+        const resourceType = filterObj.resourceType;
+        if (args[`${field}`] || args[`${field}:missing`]) {
+            const reference = `${resourceType}/` + args[`${field}`];
+            /**
+             * @type {?boolean}
+             */
+            let reference_exists_flag = null;
```

Comment:
If this is a boolean flag, could we default to False instead of null?

Reply:
no, it is tristate.  either we want records where this property is missing or not missing or either